app/controllers/users_controller.rb

```diff
@@ -0,0 +1,48 @@
+class UsersController < ApplicationController
+
+  get '/users/:slug' do
+    @user= User.find_by_slug(params[:slug])
+    erb :'users/show_user'
+  end
+
+  get '/signup' do
+    if logged_in?
+      redirect '/tweets'
+    else
+      erb :'users/create_user'
+    end
+  end
+
+  post "/signup" do
+    user = User.new(username: params[:username], email: params[:email], password: params[:password])
+		if user.save
+      session[:user_id] = user.id
+        redirect "/tweets"
+    elsif !logged_in?
+        redirect '/signup'
+    end
+  end
+
+  get '/login' do
+    if !logged_in?
+      erb :'users/login'
+    else
+      redirect "/tweets"
+    end
+  end
+
+  post "/login" do
```

Comment:
I would setup a `gate` here to make sure a user is not logged_in? before they can proceed